lib/vex/validators/uuid.ex

```diff
@@ -0,0 +1,158 @@
+      :ok
+      iex> Vex.Validators.Uuid.validate("02aa7f483ccd11e4b63e14109ff1a304", format: :default)
+      {:error, "must be a valid UUID string in default format"}
+
+  Examples when using the `:hex` option:
+
+      iex> Vex.Validators.Uuid.validate("02aa7f483ccd11e4b63e14109ff1a304", format: :hex)
+      :ok
+      iex> Vex.Validators.Uuid.validate("urn:uuid:02aa7f48-3ccd-11e4-b63e-14109ff1a304", format: :hex)
+      {:error, "must be a valid UUID string in hex format"}
+
+  Examples when using the `:urn` option:
+
+      iex> Vex.Validators.Uuid.validate("urn:uuid:02aa7f48-3ccd-11e4-b63e-14109ff1a304", format: :urn)
+      :ok
+      iex> Vex.Validators.Uuid.validate("02aa7f48-3ccd-11e4-b63e-14109ff1a304", format: :urn)
+      {:error, "must be a valid UUID string in urn format"}
+
+  ## Custom Error Messages
+
+  Custom error messages (in EEx format), provided as `:message`, can use the following values:
+
+      iex> Vex.Validators.Uuid.__validator__(:message_fields)
+      [value: "Bad value", format: "The UUID format"]
+
+  An example:
+
+      iex> Vex.Validators.Uuid.validate("not_a_uuid", format: :any,
+      ...>                                            message: "<%= value %> should be <%= format %> UUID")
+      {:error, "not_a_uuid should be any UUID"}
+  """
+
+  use Vex.Validator
+
+  @uuid_formats [:default, :hex, :urn]
+
+  @formats [:any, :not_any] ++ @uuid_formats
+
+  @urn_prefix "urn:uuid:"
+
+  @message_fields [value: "Bad value", format: "The UUID format"]
+  def validate(value, true), do: validate(value, format: :any)
+  def validate(value, false), do: validate(value, format: :not_any)
+  def validate(value, options) when options in @formats, do: validate(value, format: options)
+
+  def validate(value, options) when is_list(options) do
+    unless_skipping value, options do
+      format = options[:format]
+
+      case do_validate(value, format) do
+        :ok -> :ok
+        {:error, reason} -> {:error, message(options, reason, value: value, format: format)}
+      end
+    end
+  end
+
+  defp do_validate(<<_::64, ?-, _::32, ?-, _::32, ?-, _::32, ?-, _::96>>, :default) do
+    :ok
+  end
+
+  defp do_validate(<<_::256>>, :hex) do
+    :ok
+  end
+
+  defp do_validate(<<@urn_prefix, _::64, ?-, _::32, ?-, _::32, ?-, _::32, ?-, _::96>>, :urn) do
+    :ok
+  end
+
+  defp do_validate(_, format) when format in @uuid_formats do
+    {:error, "must be a valid UUID string in #{format} format"}
+  end
+
+  defp do_validate(value, :any) do
+    error = {:error, "must be a valid UUID string"}
+
+    Enum.reduce_while(@uuid_formats, error, fn format, _ ->
+      case do_validate(value, format) do
+        :ok -> {:halt, :ok}
+        _ -> {:cont, error}
+      end
+    end)
+  end
+
+  defp do_validate(value, :not_any) do
+    error = {:error, "must not be a valid UUID string"}
+
+    Enum.reduce_while(@uuid_formats, :ok, fn format, _ ->
```

Comment:
What about calling `do_validate(value, :any)`, this would take care of doing the reduce and make the code more DRY.